src/aap_eda/services/activation/status_manager.py

```diff
@@ -0,0 +1,104 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Module for Activation Manager."""
+
+import logging
+import typing as tp
+from functools import wraps
+
+from django.db import transaction
+
+from aap_eda.core import models
+from aap_eda.core.enums import ActivationStatus, ProcessParentType
+
+LOGGER = logging.getLogger(__name__)
+
+
+class HasDbInstance(tp.Protocol):
+    db_instance: tp.Any
+
+
+def run_with_lock(func: tp.Callable) -> tp.Callable:
```

Comment:
@Alex-Izquierdo Is this generic enough to be moved into a util like db_util and shared from there for everyone to use.

Reply:
I don't think so, at least for now. It's bound to a "has_db_instance" type which currently is only the activation_manager. In the future we might want to change, but for now I would say it is strictly related with the activation manager.

Reply:
Also, This refactor is intended to be small, only to split the status manager which I need it to be able to update the activation from the orchestrator. I would prefer to consider any other changes in different PR's.